src/Microsoft.ML.TimeSeries/IidSpikeDetector.cs

```diff
@@ -32,12 +31,12 @@ namespace Microsoft.ML.TimeSeriesProcessing
     /// <summary>
     /// This class implements the spike detector transform for an i.i.d. sequence based on adaptive kernel density estimation.
     /// </summary>
-    public sealed class IidSpikeDetector : IidAnomalyDetectionBase
+    public sealed class IidSpikeDetector : IidAnomalyDetectionBaseWrapper, IStatefulTransformer
     {
         internal const string Summary = "This transform detects the spikes in a i.i.d. sequence using adaptive kernel density estimation.";
-        public const string LoaderSignature = "IidSpikeDetector";
-        public const string UserName = "IID Spike Detection";
-        public const string ShortName = "ispike";
+        internal const string LoaderSignature = "IidSpikeDetector";
+        internal const string UserName = "IID Spike Detection";
+        internal const string ShortName = "ispike";
 
         public sealed class Arguments : TransformInputBase
```

Comment:
>Arguments [](start = 28, length = 9)

Can you rename all Arguments to Options? Should be done in other files as well.
If they are not used in the extensions, we should make all of them internal.